public/apps/configuration/panels/tenant-list/tenant-list.tsx

```diff
@@ -0,0 +1,371 @@
+  EuiPageHeader,
+  EuiPopover,
+  EuiText,
+  EuiTitle,
+  EuiGlobalToastList,
+  EuiOverlayMask,
+  EuiConfirmModal,
+} from '@elastic/eui';
+import React, { ReactNode, useEffect, useState, useCallback } from 'react';
+import { difference } from 'lodash';
+import { getAuthInfo } from '../../../../utils/auth-info-utils';
+import { AppDependencies } from '../../../types';
+import { Action, Tenant } from '../../types';
+import { renderCustomization } from '../../utils/display-utils';
+import {
+  fetchTenants,
+  transformTenantData,
+  fetchCurrentTenant,
+  resolveTenantName,
+  updateTenant,
+  requestDeleteTenant,
+  selectTenant,
+} from '../../utils/tenant-utils';
+import { getNavLinkById } from '../../../../services/chrome_wrapper';
+import { TenantEditModal } from './edit-modal';
+import { useToastState } from '../../utils/toast-utils';
+
+const PAGEID_DICT: { [key: string]: string } = {
+  DashboardId: 'dashboards',
+  VisualizationId: 'visualize',
+};
+
+export function TenantList(props: AppDependencies) {
+  const [tenantData, setTenantData] = useState<Tenant[]>([]);
+  const [errorFlag, setErrorFlag] = useState(false);
+  const [selection, setSelection] = useState<Tenant[]>([]);
+  const [isActionsPopoverOpen, setActionsPopoverOpen] = useState(false);
+  const [currentTenant, setCurrentTenant] = useState('');
+  const [currentUsername, setCurrentUsername] = useState('');
+  // Modal state
+  const [editModal, setEditModal] = useState<ReactNode>(null);
+  const [toasts, addToast, removeToast] = useToastState();
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const closeModal = () => setIsModalVisible(false);
+  const showModal = () => setIsModalVisible(true);
+
+  const fetchData = useCallback(async () => {
+    try {
+      const rawTenantData = await fetchTenants(props.coreStart.http);
+      const processedTenantData = transformTenantData(rawTenantData);
+      const activeTenant = await fetchCurrentTenant(props.coreStart.http);
+      setCurrentTenant(resolveTenantName(activeTenant, currentUsername));
+      setTenantData(processedTenantData);
+    } catch (e) {
+      console.log(e);
+      setErrorFlag(true);
+    }
+  }, [currentUsername, props.coreStart.http]);
+
+  useEffect(() => {
+    const fetchCurrentUser = async () => {
+      setCurrentUsername((await getAuthInfo(props.coreStart.http)).user_name);
+    };
+    fetchCurrentUser();
+    fetchData();
+  }, [props.coreStart.http, fetchData]);
+
+  const handleDelete = async () => {
+    const tenantsToDelete: string[] = selection.map((r) => r.tenant);
+    try {
+      await requestDeleteTenant(props.coreStart.http, tenantsToDelete);
+      setTenantData(difference(tenantData, selection));
+      setSelection([]);
+      closeModal();
+    } catch (e) {
+      console.log(e);
+    } finally {
+      setActionsPopoverOpen(false);
+    }
+  };
+
+  const handleTenantSelection = async (tenantname: string, redirectPageId: string) => {
```

Comment:
Camel case for tenantname

Reply:
not sure why using the same function for multiple use cases. Better to separate them to their own handle function for isolation logic.

Reply:
separated them to their own handle function. Also fixed camel case for tenantname